model/task/local_service.go

```diff
@@ -116,19 +143,86 @@ func (s *localService) Get(ctx context.Context, taskOpts []Task, fields ...strin
 	}
 
 	var allDBTaskResults []testresult.DbTaskTestResults
-	cur, err := s.env.DB().Collection(testresult.Collection).Find(ctx, filter, opts)
+	cur, err := s.env.CedarDB().Collection(testresult.Collection).Find(ctx, filter, opts)
 	if err != nil {
 		return nil, errors.Wrap(err, "finding DB test results")
 	}
 	if err = cur.All(ctx, &allDBTaskResults); err != nil {
 		return nil, errors.Wrap(err, "reading DB test results")
 	}
-
 	allTaskResults := make([]testresult.TaskTestResults, len(allDBTaskResults))
+
+	if len(fields) == 0 {
+		toDownload := make(chan *testresult.DbTaskTestResults, len(allDBTaskResults))
+		for i := range allDBTaskResults {
+			toDownload <- &allDBTaskResults[i]
+		}
+		close(toDownload)
+
+		config, err := evergreen.GetConfig(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "retrieving config")
+		}
+
+		var wg sync.WaitGroup
+		catcher := grip.NewBasicCatcher()
+		for i := 0; i < runtime.NumCPU(); i++ {
+			wg.Add(1)
+			go func() {
+				defer func() {
+					catcher.Add(recovery.HandlePanicWithError(recover(), nil, "test results download producer"))
+					wg.Done()
+				}()
+
+				for trs := range toDownload {
+					results, err := download(ctx, config.Buckets.Credentials, trs)
+					if err != nil {
+						catcher.Add(err)
+						return
+					}
+					trs.Results = results
+
+					if err = ctx.Err(); err != nil {
+						catcher.Add(err)
+						return
+					}
+				}
+			}()
+		}
+		wg.Wait()
+		if catcher.HasErrors() {
+			return nil, catcher.Resolve()
+		}
+	}
 	for i, dbTaskResults := range allDBTaskResults {
 		allTaskResults[i].Stats = dbTaskResults.Stats
 		allTaskResults[i].Results = dbTaskResults.Results
 	}
 
 	return allTaskResults, nil
 }
+
+// download returns a TestResult slice with the corresponding Results stored in
+// the offline blob storage. The TestResults record should be populated and the
+// environment should not be nil.
+func download(ctx context.Context, credentials evergreen.S3Credentials, t *testresult.DbTaskTestResults) ([]testresult.TestResult, error) {
+	dbTask, err := FindOneIdAndExecution(ctx, t.Info.TaskID, t.Info.Execution)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding task '%s'", t.Info.TaskID)
+	}
+	if dbTask == nil {
+		return nil, errors.Errorf("task '%s' not found", t.Info.TaskID)
+	}
+	outputInfo, ok := dbTask.GetTaskOutputSafe()
+	if !ok {
+		return nil, nil
+	}
+	return outputInfo.TestResults.downloadParquet(ctx, credentials, t)
+}
+
+func createFindQuery(id string, execution int) bson.M {
```

Comment:
nit: it's more conventional in Evergreen to call this something like `byTaskIDAndExecution`.